### PR TITLE
Merge search-by-location into Master

### PR DESCRIPTION
**Pull request by sehoven**
Wrote searching tests and search by location functionality.

Solves issues [#101](https://github.com/CMPUT301F16T01/Carrier/issues/101) and [#153](https://github.com/CMPUT301F16T01/Carrier/issues/153)

### CODE REVIEW

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 1]
```diff
@@ -2,55 +2,34 @@
...
-
-        if (result == null) {
-            System.out.print( "null line" );
-        }
-
-        assertTrue( "Failed to log in for test.", uc.logInUser( loggedInUser.getUsername() ) );
-    }
-
     // This tear down method may not be working entirely as expected...test further
     protected void tearDown() throws Exception {
```
Comment (Hreherch) [inline app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:36]: Is it working as expected?
Can you add a comment as to what your tearDown() method does?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java [hunk 2]
```diff
@@ -115,6 +114,108 @@ protected RequestList doInBackground(String... search_parameters) {
...
+     *             { "match": { "status": 2 }}
+     *           ]
+     *         }
+     *       },
+     *       "filter" : {
+     *         "geo_distance" : {
+     *           "distance" : "50km",
+     *           "location" : [-113, 52]
+     *         }
+     *       }
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Can we keep these somewhere else, like on the Wiki?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 3]
```diff
@@ -151,12 +152,19 @@ public void searchByKeyword(String keyword) {
...
 
-        /**
+    /**
      * Search requests by a location. This sets it so the singleton contains the information for
      * this query. Use getResults() to get the information.
      */
-    public void searchByLocation( /* location parameters? */ ) {
-
+    public void searchByLocation(Location location) {
+        // TODO check how these are sorted, we want to sort them by those closest to those furthest away
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Did you complete this TODO while working on this branch? (I remember you mentioning it)

File: app/src/main/java/comcmput301f16t01/github/carrier/UserController.java [hunk 4]
```diff
@@ -211,7 +211,7 @@ public static String createNewUser(String username, String email, String phoneNu
         }
 
         if (!Patterns.PHONE.matcher(phoneNumber).matches()) {
-            return "That doesn't look liek a valid phone number!";
+            return "That doesn't look like a valid phone number!";
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/UserController.java]: 💁 My bad, thank you for catching this!

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java [hunk 5]
```diff
@@ -72,61 +51,63 @@ private void chillabit( long time ) {
         }
     }
 
-    // TODO implement searchByLocation, this test will not pass
     public void testDriverSearchByLocation() {
```
Comment (Hreherch) [inline app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: In comments could I have what you are testing, and if you test for any exceptional cases? 
Also maybe include reference to which use case this tests, if applicable.

**Review by Hreherch — CHANGES_REQUESTED** [review]
I'm not sure if I can confirm/deny that the "null location" issue is correct, do you happen to have a test for it that I missed? (Not a necessary component to complete for approval) 

Otherwise doing great! Just some minor things to potentially change.

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java (hunk 1 above)
Comment (sehoven) [reply app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:36]: Created an issue to implement the fix from the singleton-refactor branch once both are merged to master.

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java (hunk 2 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Moved to Google Docs.

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java (hunk 5 above)
Comment (sehoven) [reply app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: Added use case references.

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java (hunk 3 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Thank you. I removed this TODO. This is done.

**Comment by sehoven** [comment]
The null location fix is in SetLocationsActivity.java in the getCurrentLocation() method. I check if lastLocation is null (which only happens if there was a problem retrieving the current location). If it is null, it defaults to (0, 0) (which is somewhere in the middle of the ocean).

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java [hunk 6]
```diff
@@ -129,8 +129,83 @@ protected RequestList doInBackground(String... search_parameters) {
...
+            return foundRequests;
+        }
+    }
+
+    /**
+     * Get all of a rider's requests including a filter by status
+=======
      * Get all of a rider's requests filtered by status
      * @see RequestController#fetchRequestsWhereRider(User, Integer...)
+>>>>>>> 92fda80c294f949b664f49d206e3dedeea1f17a7
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Git decided to be awkward.

Comment (Hreherch) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: You don't have to fix this, it's in #167, I think. 
I'll approve this branch when you merge that with it's changes?

**Comment by Hreherch** [comment]
The only minor issue I can find is that I can offer twice on the same request if I haven't gone all the way back to the search buttons, this is more of an enhancement at this point. 
Also "Search Results s" is one of the titles. But that's more of a quality issue :P

**Review by Hreherch — APPROVED** [review]

**Pull request merged**